bofire/data_models/strategies/samplers/universal_constraint.py

```diff
@@ -0,0 +1,47 @@
+from typing import Literal, Type
+
+from bofire.data_models.constraints.api import (
+    LinearEqualityConstraint,
+    LinearInequalityConstraint,
+    NChooseKConstraint,
+    NonlinearEqualityConstraint,
+    NonlinearInequalityConstraint,
+)
+from bofire.data_models.features.api import (
+    ContinuousInput,
+    ContinuousOutput,
+    Feature,
+)
+from bofire.data_models.strategies.samplers.sampler import SamplerStrategy
+
+
+class UniversalConstraintSampler(SamplerStrategy):
+    """Sampler that generates samples by optimization in IPOPT.
+
+    Attributes:
+        domain (Domain): Domain defining the constrained input space
+        sampling_fraction (float, optional): Fraction of sampled points to total points generated in
+            the sampling process. Defaults to 0.3.
+        ipopt_options (dict, optional): Dictionary containing options for the IPOPT solver. Defaults to {"maxiter":200, "disp"=0}.
+    """
+
+    type: Literal["UniversalConstraintSampler"] = "UniversalConstraintSampler"
+    sampling_fraction: float = 0.3
```

Comment:
```suggestion
    sampling_fraction: Annotated[float, Field(gt=0, lt=1)] = 0.3
```

By this we have direct validation that the sampling fraction is between 0 and 1 as soons as we switch to pydantic2 which will be soon ;)